### PR TITLE
Create ISmartWallet and integrate Alchemy Light Account

### PR DESCRIPTION
**Pull request by Yorkemartin**
This PR outlines a potential architecture:

* All accounts must implement the interface ISmartWallet, which contains all the methods we expect users to call on any given account, independent of its underlying code
* AlchemySmartWallet extends Light Account, and implements ISmartWallet
* FunSmartWallet extends the old SmartWallet entity, and Implementation requirement of ISmartWallet
* Any future account's by any other providers will have to implement the ISmartWallet interface
* In SnowBall.ts, a smart wallet is defined as type ISmartWallet

needs: 
* more test coverage

out of scope: 
* removing SmartWallet.ts
* implementing FunSmartWallet the right way

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,3 +36,4 @@ next-env.d.ts
 
 yarn.lock
 .env
+package-lock.json
```
Comment (ericlewis) [inline .gitignore:39]: package-locks should be part of the repo

File: package.json [hunk 2]
```diff
@@ -50,9 +50,10 @@
     "alchemy-sdk": "^2.10.0",
     "ethers": "^5.5.1",
     "tsdom": "^0.6.9",
-    "viem": "^1.9.3"
+    "viem": "^1.19.8"
   },
   "devDependencies": {
+    "@alchemy/aa-accounts": "^1.2.0",
```
Comment (ericlewis) [inline package.json:56]: this should be a regular dependency

File: src/wallet/AlchemySmartWalletV2.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    });
+    this.provider = provider;
+    this.auth = auth;
+    this.smartWalletProviderInfo = smartWalletProviderInfo;
+  }
+
+  switchChain(): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
+  sendUserOperation(
```
Comment (ericlewis) [inline src/wallet/AlchemySmartWalletV2.ts:48]: just a nit, but new lines between on these methods would be good

**Review by ericlewis — COMMENTED** [review]
the interface approach is good, this looks alright as a start.

File: src/wallet/AlchemySmartWalletV2.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  UserOperationReceipt,
+  UserOperationResponse,
+} from "@alchemy/aa-core";
+import { LightSmartContractAccount } from "@alchemy/aa-accounts";
+import { viemChain } from "../helpers/chains";
+import { Address, Hex } from "viem";
+import { getDefaultLightAccountFactoryAddress } from "@alchemy/aa-accounts";
+import { ISmartWalletV2 } from "./ISmartWalletV2";
+import { PKPEthersWallet } from "@lit-protocol/pkp-ethers";
+export class AlchemySmartWalletV2
```
Comment (VPhung24) [inline src/wallet/AlchemySmartWalletV2.ts:16]: should we use V2 when the SDK is still in alpha? devs should expect major changes

File: src/wallet/FunSmartWalletV2.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable */
+import type { Address } from "viem";
+import { UserOperationResponse, UserOperationReceipt } from "@alchemy/aa-core";
+import { SmartWallet } from "./SmartWallet";
+import type { ISmartWalletV2 } from "./ISmartWalletV2";
+
+export class FunSmartWallet extends SmartWallet implements ISmartWalletV2 {
```
Comment (VPhung24) [inline src/wallet/FunSmartWalletV2.ts:7]: again on V2: don't think we have users on V1 to need

File: src/wallet/AlchemySmartWalletV2.ts (hunk 4 above)
Comment (Yorkemartin) [reply src/wallet/AlchemySmartWalletV2.ts:16]: agreed

**Pull request merged**